src/modules/Elsa.Workflows.Core/VariableStorageDrivers/WorkflowInstanceStorageDriver.cs

```diff
@@ -39,7 +40,15 @@ public ValueTask WriteAsync(string id, object value, StorageDriverContext contex
     {
         var dictionary = GetVariablesDictionary(context);
         var node = dictionary.GetValueOrDefault(id);
-        return new(node);
+        var variable = context.Variable;
+        var variableType = variable.GetVariableType();
+        var options = new ObjectConverterOptions
+        {
+            DeserializeJsonObjectToObject = true,
+            SerializerOptions = payloadSerializer.GetOptions()  
```

Comment:
Could we just inject the options instead of the entire payloadSerializer if that is all that we are using?

Reply:
Those options aren’t injectable, but if you have something in mind I’m all ears.